plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -938,10 +942,27 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
-            provider = new STSAssumeRoleSessionCredentialsProvider.Builder(iamRole, "trino-session")
-                    .withExternalId(externalId)
-                    .withLongLivedCredentialsProvider(provider)
-                    .build();
+            String stsEndpointOverwrite = conf.get(STS_ENDPOINT);
```

Comment:
I'd format this block a bit differently, `withLongLivedCredentialsProvider` is also deprecated so using `AWSSecurityTokenServiceClientBuilder` is preferred in all cases:

```
String stsEndpointOverwrite = conf.get(STS_ENDPOINT);
String stsRegionOverwrite = conf.get(STS_REGION);

STSAssumeRoleSessionCredentialsProvider.Builder providerWithRole = new STSAssumeRoleSessionCredentialsProvider.Builder(iamRole, "trino-session")
    .with(<all the common properties>);
AWSSecurityTokenServiceClientBuilder stsClient = ...
    .withCredentials(provider);

if (<both overrides are set>) {
    stsClient.withEndpointConfiguration(...);
}
else if (<region is set>) {
    stsClient.withRegion(...);
}

provider = providerWithRole
    .withStsClient(stsClient.build())
    .build();
```

Reply:
I thought the same and wanted to get rid of the deprecated `withLongLivedCredentialsProvider` as well, but that resulted in errors in `TrinoS3FileSystemTest`: `com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.` 

So the question is: fix the tests setup or stay with the deprecated method (or use a option I have not thought of) ?

Reply:
You probably need to use the default region provider chain, and if that comes up empty use `us-east-1`

Reply:
Good point, I updated the PR accordingly